src/main/java/com/ericsson/vici/Fetcher.java

```diff
@@ -143,60 +146,64 @@ public Events fetchEvents(Preferences preferences) {
         log.info("Downloading eiffel-events from: " + preferences.getUrl());
 
         RestTemplate restTemplate = new RestTemplate();
-        ResponseEntity<EiffelEvent[]> responseEntity;
-        EiffelEvent[] eiffelEvents = null;
+        ResponseEntity<EiffelEREventsResponse> responseEntity;
+        ArrayList<EiffelEvent> eiffelEvents = new ArrayList<>();
 
         Pattern pattern = Pattern.compile("^localFile\\[(.+)]$");
         Matcher matcher = pattern.matcher(preferences.getUrl().trim());
 
         long eventsFetchedAt = System.currentTimeMillis();
 
-        if (matcher.find()) {
-//            System.out.println("Request for local file " + matcher.group(1) + ".json");
-//            responseEntity = restTemplate.getForEntity("http://127.0.0.1:8080/" + matcher.group(1), EiffelEvent[].class);
-
+        if (matcher.find()) { // Local file
             ObjectMapper mapper = new ObjectMapper();
             Resource resource = new ClassPathResource("static/assets/" + matcher.group(1) + ".json");
             InputStream jsonFileStream;
 
             try {
                 jsonFileStream = resource.getInputStream();
-                eiffelEvents = mapper.readValue(jsonFileStream, EiffelEvent[].class);
+                EiffelEvent[] eiffelEventList = mapper.readValue(jsonFileStream, EiffelEvent[].class);
+                Collections.addAll(eiffelEvents, eiffelEventList);
             } catch (IOException e) {
                 e.printStackTrace();
             }
-        } else {
-            Query query = new Query(null, null, 0, Integer.MAX_VALUE, false, null, true);
-            ObjectMapper mapper = new ObjectMapper();
-            JSONObject queryJson = null;
-            try {
-                queryJson = new JSONObject(mapper.writeValueAsString(query));
-            } catch (JsonProcessingException e) {
-                e.printStackTrace();
-            }
+        } else { // Eiffel ER Rest API version??
+            URIBuilder builder;
+            int totalNumberItems = Integer.MAX_VALUE;
+//            int pageSize = 59923; // seems to be the maximum of events the Eiffel ER can handle at the moment.
+            int pageSize = 2000;
+            int page = 1;
+
+            while (eiffelEvents.size() < totalNumberItems) {
```

Comment:
what happens if you have 50 pages and 100000 events and while you fetch the 5th page 10 events will come in? totalNumberOfItems will be 100010 and the latest 10 will show up in the first page not last. The loop will loop for ever since you will never read the last 10 events that came after reading the first page. the events size will be 10 less than the total number of items that you fetch continuously from each response.

Reply:
If the Eiffel-ER works as expected that should be handled by the `totalNumberItems = response.totalNumberItems;` which is updated every loop. 
But yes there is a problem here if the number of events changes in the event repository while Vici is fetching the events.
But at least this makes it possible to start testing the eiffel-er with vici for now. A temporary solution. Will be fixed in next commit.